pkg/client/typed/discovery/discovery_client.go

```diff
@@ -212,6 +212,9 @@ func setDiscoveryDefaults(config *restclient.Config) error {
 	config.APIPath = ""
 	config.GroupVersion = nil
 	config.Codec = runtime.NoopEncoder{api.Codecs.UniversalDecoder()}
+	if config.UserAgent == "" {
```

Comment:
Things in IF blocks are important.  You need to instantly be able to assess
what the IF block does before you can accurately review the code.  Code
conventions are intended to make code more readable.  Having a consistent
way to represent empty string comparison makes code more readable.  We have
enforced that convention elsewhere, we just aren't enforcing it
consistently.  Enforcing it consistently is a net benefit to the code.

On Thu, Mar 10, 2016 at 7:49 AM, David Eads <notifications@github.com>
wrote:

> In pkg/client/typed/discovery/discovery_client.go
> <https://github.com/kubernetes/kubernetes/pull/22746#discussion_r55673650>
> :
>
> > @@ -212,6 +212,9 @@ func setDiscoveryDefaults(config *restclient.Config) error {
> >  	config.APIPath = ""
> >  	config.GroupVersion = nil
> >  	config.Codec = runtime.NoopEncoder{api.Codecs.UniversalDecoder()}
> > +	if config.UserAgent == "" {
>
> As per Russ cox, it does not matter it seems:
>
> I might choose a different windmill... :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/22746/files#r55673650>.
>